src/data/zmod/basic.lean

```diff
@@ -715,6 +715,9 @@ begin
   { simp only [neg_eq_self_mod_two, nat.cast_succ, int.nat_abs, int.cast_neg_succ_of_nat] }
 end
 
+@[simp]
+lemma add_self_mod_two : ∀ (m : zmod 2), m + m = 0 := dec_trivial
```

Comment:
Please generalize this to `[ring R] [char_p R 2]`.

Reply:
It turns out that this more general version already exists in a file that I had not imported:

```lean
--  in algebra.char_p.two

lemma add_self_eq_zero (x : R) : x + x = 0 :=
by rw [←two_smul R x, two_eq_zero, zero_smul]
```

Reply:
Also, `R` is a semiring.